modules/db/src/main/scala/gem/dao/package.scala

```diff
@@ -49,8 +49,8 @@ package object dao extends MoreTupleOps with ToUserProgramRoleOps {
   implicit val InstantMeta: Meta[Instant] =
     Meta[Timestamp].nxmap(_.toInstant, Timestamp.from)
 
-  implicit val LocationMeta: Meta[Location] =
-    Meta[List[Int]].nxmap(Location.fromList, _.toList)
+  implicit val LocationMeta: Meta[Location.Middle] =
+    Meta[List[Int]].nxmap(Location.unsafeMiddle(_), _.toList)
```

Comment:
From the doc it's an "invariant map".  I would appreciate a theoretical explanation of that myself.  @tpolecat would have a [better explanation](https://github.com/tpolecat/tpolecat.github.io/blob/master/_doobie-0.2.1/10-Custom-Mappings.md), but this is how I understand it in imprecise language:

We're telling doobie how to translate between `Location.Middle` classes and database types, which it can do because it already knows how to handle `List`.  The the 'n' is a reference to the fact that you can supply functions that don't worry about `null` values because if it sees a `null` it will return a `null`.  With this defined, doobie can return values from queries where a `Location.Middle` is expected and can accept a `Location.Middle` to insert in a row.